### DISCUSSION TITLE
Map sub data provider from parent provider json array result

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3326] [Help]
I have json result from rest datasource as below and my design structure look like this:

![image](https://user-images.githubusercontent.com/597759/141243578-7d3ff72b-9f7c-46d1-8301-f940e1e329e8.png)

Is it possible to set data for Sub Data Provider map to requestBodyDictionaries . 

REST Datasource
![image](https://user-images.githubusercontent.com/597759/141243468-1b6e01c1-9865-4fd3-9608-e18816121bc7.png)

**Comment by mjashanks** [comment]
Hey  @ryzam - have you tried a transformer? https://docs.budibase.com/building-apps/data/transformers 

e.g.

```javascript 
const dictionaries = [];

for (let row of data) {
    for (let dictionary of row.requestBodyDictionaries) {
        dictionaries.push(dictionary);
    }
}

return dictionaries;
```

This will make Sub Data Provider return `requestBodyDictionaries`, all in one array

**Reply by ryzam** [reply]
Hi @mjashanks , thank you for your answer. I'm aware about transformer but using transformer in this case will make Sub Data Provider call new REST Data source for every row. I'm thinking if we can set Sub Data Provider to get in memory of data from his parent data provider.

**Reply by mjashanks** [reply]
Got it. We are actually working on this at the moment: https://github.com/Budibase/budibase/issues/3173

It's at the PoC stage - but it looks like you will be able to add a data provider, that you can attach to an array (thus no API call).

We will likely also add a second "block" component, which is a `Data Provider + Repeater` - which will save a lot of necessary nesting!
